compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1129,9 +1129,17 @@ fn should_encode_trait_impl_trait_tys(tcx: TyCtxt<'_>, def_id: DefId) -> bool {
     })
 }
 
-// Return `false` to avoid encoding impl trait in trait, while we don't use the query.
-fn should_encode_fn_impl_trait_in_trait<'tcx>(_tcx: TyCtxt<'tcx>, _def_id: DefId) -> bool {
-    false
+// We only encode impl trait in trait when using `lower-impl-trait-in-trait-to-assoc-ty` unstable
+// option.
+fn should_encode_fn_impl_trait_in_trait<'tcx>(tcx: TyCtxt<'tcx>, def_id: DefId) -> bool {
+    if tcx.sess.opts.unstable_opts.lower_impl_trait_in_trait_to_assoc_ty {
+        match tcx.opt_associated_item(def_id) {
```

Comment:
hm, this probably should check for only functions too... I was thinking of something simpler than this.

```
if tcx.sess.opts.unstable_opts.lower_impl_trait_in_trait_to_assoc_ty
    && let Some(assoc_item) = tcx.opt_associated_item(def_id)
    && assoc_item.container == ty::AssocItemContainer::TraitContainer
    && assoc_item.kind == ty::AssocKind::Fn
{
    true
} else {
    false
}
```